BUG: Build fails on 4.0.0

Using Cmake with Ninja, Fedora based OS. All deps installed including  vulkan-headers and glslc.

[95/368] Building CXX object lib/framework/CMakeFiles/framework.dir/wzstring.cpp.o
FAILED: lib/framework/CMakeFiles/framework.dir/wzstring.cpp.o 
/usr/bin/c++ -DHAVE_CONFIG_H -DJSON_USE_IMPLICIT_CONVERSIONS=0 -DUTF8PROC_STATIC -DYY_NO_UNISTD_H -I. -I../. -I../3rdparty -I../3rdparty/date/include -I../3rdparty/utf8proc/. -isystem ../3rdparty/glm -fstack-protector-strong -fstack-clash-protection -Wpedantic -Wall -Wextra -fno-common -fno-math-errno -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Winit-self -Woverloaded-virtual -Wstrict-null-sentinel -Wwrite-strings -Wpointer-arith -Wmissing-noreturn -Wnon-virtual-dtor -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca-larger-than=1024 -Wrestrict -Wfloat-conversion -Wstringop-truncation -Werror -Wno-logical-op -Wno-float-conversion -Wno-sign-compare -Wno-unused-parameter -Wno-format-truncation -DNDEBUG -fPIC -std=c++11 -MD -MT lib/framework/CMakeFiles/framework.dir/wzstring.cpp.o -MF lib/framework/CMakeFiles/framework.dir/wzstring.cpp.o.d -o lib/framework/CMakeFiles/framework.dir/wzstring.cpp.o -c ../lib/framework/wzstring.cpp
In file included from ../lib/framework/frame.h:26,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/wzstring.cpp: In constructor 'WzString::WzString(const char*, int)':
../lib/framework/wzstring.cpp:93:71: error: 'numeric_limits' is not a member of 'std'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                       ^~~~~~~~~~~~~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:93:86: error: expected primary-expression before 'int'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                                      ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:93:86: error: expected ')' before 'int'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                                      ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:93:86: error: expected ')' before 'int'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                                      ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
../lib/framework/wzglobal.h:688:25: note: to match this '('
  688 | # define likely(expr)   (expr)
      |                         ^
../lib/framework/debug.h:117:29: note: in expansion of macro 'likely'
  117 |         do { bool _wzeval = likely(expr); if (!_wzeval) { ASSERT_FAILURE(expr, #expr, AT_MACRO, __FUNCTION__, __VA_ARGS__); return retval; } } while (0)
      |                             ^~~~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
In file included from ../lib/framework/string_ext.h:24,
                 from ../lib/framework/frame.h:44,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/wzstring.cpp:93:171: error: 'numeric_limits' is not a member of 'std'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                                                                                                                           ^~~~~~~~~~~~~~
../lib/framework/debug.h:71:54: note: in definition of macro 'ASSERT_FAILURE'
   71 |           (void)_debug(__LINE__, LOG_INFO, function, __VA_ARGS__), \
      |                                                      ^~~~~~~~~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:93:186: error: expected primary-expression before 'int'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                                                                                                                                                                                          ^~~
../lib/framework/debug.h:71:54: note: in definition of macro 'ASSERT_FAILURE'
   71 |           (void)_debug(__LINE__, LOG_INFO, function, __VA_ARGS__), \
      |                                                      ^~~~~~~~~~~
../lib/framework/wzstring.cpp:93:17: note: in expansion of macro 'ASSERT_OR_RETURN'
   93 |                 ASSERT_OR_RETURN(, strLen <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", strLen, std::numeric_limits<int>::max());
      |                 ^~~~~~~~~~~~~~~~
In file included from ../lib/framework/frame.h:26,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/wzstring.cpp: In static member function 'static WzString WzString::fromUtf8(const string&)':
../lib/framework/wzstring.cpp:114:79: error: 'numeric_limits' is not a member of 'std'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                               ^~~~~~~~~~~~~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:114:94: error: expected primary-expression before 'int'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                                              ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:114:94: error: expected ')' before 'int'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                                              ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:114:94: error: expected ')' before 'int'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                                              ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzglobal.h:688:25: note: to match this '('
  688 | # define likely(expr)   (expr)
      |                         ^
../lib/framework/debug.h:117:29: note: in expansion of macro 'likely'
  117 |         do { bool _wzeval = likely(expr); if (!_wzeval) { ASSERT_FAILURE(expr, #expr, AT_MACRO, __FUNCTION__, __VA_ARGS__); return retval; } } while (0)
      |                             ^~~~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
In file included from ../lib/framework/string_ext.h:24,
                 from ../lib/framework/frame.h:44,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/wzstring.cpp:114:185: error: 'numeric_limits' is not a member of 'std'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                                                                                                                                         ^~~~~~~~~~~~~~
../lib/framework/debug.h:71:54: note: in definition of macro 'ASSERT_FAILURE'
   71 |           (void)_debug(__LINE__, LOG_INFO, function, __VA_ARGS__), \
      |                                                      ^~~~~~~~~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:114:200: error: expected primary-expression before 'int'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |                                                                                                                                                                                                        ^~~
../lib/framework/debug.h:71:54: note: in definition of macro 'ASSERT_FAILURE'
   71 |           (void)_debug(__LINE__, LOG_INFO, function, __VA_ARGS__), \
      |                                                      ^~~~~~~~~~~
../lib/framework/wzstring.cpp:114:9: note: in expansion of macro 'ASSERT_OR_RETURN'
  114 |         ASSERT_OR_RETURN(WzString(), str.length() <= static_cast<size_t>(std::numeric_limits<int>::max()), "String length (%zu) exceeds maximum support length: %d", str.length(), std::numeric_limits<int>::max());
      |         ^~~~~~~~~~~~~~~~
../lib/framework/wzstring.cpp: In member function 'int WzString::length() const':
../lib/framework/wzstring.cpp:188:33: error: 'numeric_limits' is not a member of 'std'
  188 |         if (len_distance > std::numeric_limits<int>::max())
      |                                 ^~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:188:48: error: expected primary-expression before 'int'
  188 |         if (len_distance > std::numeric_limits<int>::max())
      |                                                ^~~
../lib/framework/wzstring.cpp:188:48: error: expected ')' before 'int'
  188 |         if (len_distance > std::numeric_limits<int>::max())
      |            ~                                   ^~~
      |                                                )
In file included from ../lib/framework/frame.h:26,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/wzstring.cpp:190:45: error: 'numeric_limits' is not a member of 'std'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                                             ^~~~~~~~~~~~~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/debug.h:109:9: note: in expansion of macro 'ASSERT_HELPER'
  109 |         ASSERT_HELPER(expr, AT_MACRO, __FUNCTION__, __VA_ARGS__)
      |         ^~~~~~~~~~~~~
../lib/framework/wzstring.cpp:190:17: note: in expansion of macro 'ASSERT'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                 ^~~~~~
../lib/framework/wzstring.cpp:190:60: error: expected primary-expression before 'int'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                                                            ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/debug.h:109:9: note: in expansion of macro 'ASSERT_HELPER'
  109 |         ASSERT_HELPER(expr, AT_MACRO, __FUNCTION__, __VA_ARGS__)
      |         ^~~~~~~~~~~~~
../lib/framework/wzstring.cpp:190:17: note: in expansion of macro 'ASSERT'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                 ^~~~~~
../lib/framework/wzstring.cpp:190:60: error: expected ')' before 'int'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                                                            ^~~
../lib/framework/wzglobal.h:688:26: note: in definition of macro 'likely'
  688 | # define likely(expr)   (expr)
      |                          ^~~~
../lib/framework/debug.h:109:9: note: in expansion of macro 'ASSERT_HELPER'
  109 |         ASSERT_HELPER(expr, AT_MACRO, __FUNCTION__, __VA_ARGS__)
      |         ^~~~~~~~~~~~~
../lib/framework/wzstring.cpp:190:17: note: in expansion of macro 'ASSERT'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                 ^~~~~~
../lib/framework/wzglobal.h:688:25: note: to match this '('
  688 | # define likely(expr)   (expr)
      |                         ^
../lib/framework/debug.h:95:11: note: in expansion of macro 'likely'
   95 |           likely(expr) ? /* if (expr) */ \
      |           ^~~~~~
../lib/framework/debug.h:109:9: note: in expansion of macro 'ASSERT_HELPER'
  109 |         ASSERT_HELPER(expr, AT_MACRO, __FUNCTION__, __VA_ARGS__)
      |         ^~~~~~~~~~~~~
../lib/framework/wzstring.cpp:190:17: note: in expansion of macro 'ASSERT'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                 ^~~~~~
../lib/framework/wzstring.cpp:190:164: error: expected ')' before ';' token
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                                                                                                                                                                    ^
In file included from ../lib/framework/string_ext.h:24,
                 from ../lib/framework/frame.h:44,
                 from ../lib/framework/wzstring.cpp:20:
../lib/framework/debug.h:94:9: note: to match this '('
   94 |         ( \
      |         ^
../lib/framework/debug.h:109:9: note: in expansion of macro 'ASSERT_HELPER'
  109 |         ASSERT_HELPER(expr, AT_MACRO, __FUNCTION__, __VA_ARGS__)
      |         ^~~~~~~~~~~~~
../lib/framework/wzstring.cpp:190:17: note: in expansion of macro 'ASSERT'
  190 |                 ASSERT(len_distance <= std::numeric_limits<int>::max(), "The length (in codepoints) of a WzString exceeds the max of the return type for length()");
      |                 ^~~~~~
../lib/framework/wzstring.cpp:191:29: error: 'numeric_limits' is not a member of 'std'
  191 |                 return std::numeric_limits<int>::max();
      |                             ^~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:191:44: error: expected primary-expression before 'int'
  191 |                 return std::numeric_limits<int>::max();
      |                                            ^~~
../lib/framework/wzstring.cpp:191:44: error: expected ';' before 'int'
  191 |                 return std::numeric_limits<int>::max();
      |                                            ^~~
      |                                            ;
../lib/framework/wzstring.cpp:191:47: error: expected unqualified-id before '>' token
  191 |                 return std::numeric_limits<int>::max();
      |                                               ^
../lib/framework/wzstring.cpp: In static member function 'static WzString WzString::number(double)':
../lib/framework/wzstring.cpp:504:38: error: 'numeric_limits' is not a member of 'std'
  504 |         ss << std::setprecision(std::numeric_limits<double>::digits10) << n;
      |                                      ^~~~~~~~~~~~~~
../lib/framework/wzstring.cpp:504:53: error: expected primary-expression before 'double'
  504 |         ss << std::setprecision(std::numeric_limits<double>::digits10) << n;
      |                                                     ^~~~~~
[100/368] Building CXX object lib/ivis_opengl/CMakeFiles/ivis-opengl.dir/gfx_api_gl.cpp.o
ninja: build stopped: subcommand failed.

--- a/lib/framework/frame.cpp
+++ b/lib/framework/frame.cpp
@@ -34,6 +34,8 @@
 #include "frameresource.h"
 #include "input.h"
 
+#include <limits>
+
 /************************************************************************************
  *
  *	Player globals

--- a/lib/framework/wzconfig.cpp
+++ b/lib/framework/wzconfig.cpp
@@ -27,6 +27,7 @@
 #include <physfs.h>
 #include "file.h"
 #include <sstream>
+#include <limits>
 #include "physfs_ext.h"
 
 WzConfig::~WzConfig()

--- a/lib/framework/wzstring.cpp
+++ b/lib/framework/wzstring.cpp
@@ -23,6 +23,7 @@
 #include <cctype>
 #include <sstream>
 #include <iomanip>
+#include <limits>
 #include <utfcpp/source/utf8.h>
 #include <utf8proc/utf8proc.h>
 

--- a/lib/ivis_opengl/gfx_api_gl.cpp
+++ b/lib/ivis_opengl/gfx_api_gl.cpp
@@ -30,6 +30,7 @@
 #include <unordered_set>
 #include <unordered_map>
 #include <regex>
+#include <limits>
 
 #include <glm/gtc/type_ptr.hpp>
 

--- a/lib/ivis_opengl/gfx_api_vk.cpp
+++ b/lib/ivis_opengl/gfx_api_vk.cpp
@@ -47,6 +47,7 @@
 #include <set>
 #include <unordered_set>
 #include <map>
+#include <limits>
 
 // Fix #define MemoryBarrier coming from winnt.h
 #undef MemoryBarrier

--- a/lib/ivis_opengl/screen.cpp
+++ b/lib/ivis_opengl/screen.cpp
@@ -52,6 +52,7 @@
 #include <string>
 #include <vector>
 #include <cstring>
+#include <limits>
 #ifndef GLM_ENABLE_EXPERIMENTAL
 	#define GLM_ENABLE_EXPERIMENTAL
 #endif

--- a/lib/ivis_opengl/textdraw.cpp
+++ b/lib/ivis_opengl/textdraw.cpp
@@ -49,6 +49,7 @@ static float font_colour[4] = {1.f, 1.f, 1.f, 1.f};
 #include "ft2build.h"
 #include <unordered_map>
 #include <memory>
+#include <limits>
 
 #if defined(HB_VERSION_ATLEAST) && HB_VERSION_ATLEAST(1,0,5)
 //	#define WZ_FT_LOAD_FLAGS (FT_LOAD_DEFAULT | FT_LOAD_TARGET_LCD) // Needs further testing on low-DPI displays

--- a/lib/netplay/netplay.cpp
+++ b/lib/netplay/netplay.cpp
@@ -41,6 +41,7 @@
 #include <memory>
 #include <thread>
 #include <atomic>
+#include <limits>
 #include <sodium.h>
 
 #include "netplay.h"

--- a/lib/netplay/nettypes.cpp
+++ b/lib/netplay/nettypes.cpp
@@ -40,6 +40,7 @@
 #include "netlog.h"
 #include "src/order.h"
 #include <cstring>
+#include <limits>
 
 /// There is a game queue representing each player. The game queues are synchronised among all players, so that all players process the same game queue
 /// messages at the same game time. The game queues should be used, even in single-player. Players should write to their own queue, not to other player's

--- a/lib/sound/playlist.cpp
+++ b/lib/sound/playlist.cpp
@@ -30,6 +30,7 @@
 #include <algorithm>
 #include <unordered_map>
 #include <sstream>
+#include <limits>
 
 #include <optional-lite/optional.hpp>
 using nonstd::optional;

--- a/src/frontend.cpp
+++ b/src/frontend.cpp
@@ -39,6 +39,8 @@
 #include "lib/widget/slider.h"
 #include "lib/widget/dropdown.h"
 
+#include <limits>
+
 #include "advvis.h"
 #include "challenge.h"
 #include "component.h"

--- a/src/integrations/wzdiscordrpc.cpp
+++ b/src/integrations/wzdiscordrpc.cpp
@@ -35,6 +35,7 @@
 #include <chrono>
 #include <functional>
 #include <memory>
+#include <limits>
 #include <discord_rpc.h>
 #include <sodium.h>
 #include <EmbeddedJSONSignature.h>

--- a/src/notifications.cpp
+++ b/src/notifications.cpp
@@ -35,6 +35,7 @@
 #include "init.h"
 #include "frend.h"
 #include <algorithm>
+#include <limits>
 
 // MARK: - Notification Ignore List
 

--- a/src/qtscript.cpp
+++ b/src/qtscript.cpp
@@ -67,6 +67,7 @@
 #include <sstream>
 #include <iomanip>
 #include <queue>
+#include <limits>
 
 #include "wzscriptdebug.h"
 #include "quickjs_backend.h"

--- a/src/quickjs_backend.cpp
+++ b/src/quickjs_backend.cpp
@@ -77,6 +77,7 @@
 #include <unordered_set>
 #include "lib/framework/file.h"
 #include <unordered_map>
+#include <limits>
 
 #if !defined(__clang__) && defined(__GNUC__) && __GNUC__ >= 8
 #pragma GCC diagnostic push

--- a/src/visibility.cpp
+++ b/src/visibility.cpp
@@ -30,6 +30,8 @@
 #include "lib/sound/audio_id.h"
 #include "lib/ivis_opengl/ivisdef.h"
 
+#include <limits>
+
 #include "visibility.h"
 
 #include "objects.h"

--- a/src/wzscriptdebug.cpp
+++ b/src/wzscriptdebug.cpp
@@ -64,6 +64,7 @@
 
 #include <numeric>
 #include <algorithm>
+#include <limits>
 
 static std::shared_ptr<W_SCREEN> debugScreen = nullptr;
 static std::shared_ptr<WZScriptDebugger> globalDialog = nullptr;

